applicationset/controllers/applicationset_controller.go

```diff
@@ -139,6 +145,19 @@ func (r *ApplicationSetReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 			}
 			logCtx.Debugf("ownerReferences referring %s is deleted from generated applications", appsetName)
 		}
+		if isRollingSyncDeletionReversed(&applicationSetInfo) {
```

Comment:
nitpick: Can we have a simpler and easy to understand way to check this ? something like `isRollingSync() && isDeletionOrderReverse()`

Reply:
does renaming the function to `isProgressiveSyncDeletionOrderReversed` help?

This function also checks for the length of the `appset.Spec.Strategy.RollingSync.Steps` as well, so simplifying it as per your suggestion would be 
`progressiveSyncsRollingSyncStrategyEnabled() && isDeletionOrderReverse ()` 
[instead of `isRollingSync() && isDeletionOrderReverse()`] which isn't very easy to read imo. wdyt?